### PR TITLE
fix: task recent runs should filter by version

### PR DESCRIPTION
**Pull request by ursucarina**
# TL;DR
filters tasks by version

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
filters tasks by version

test on:
[wf1](https://development.uniondemo.run/console/projects/flytesnacks/domains/development/workflows/async_prototype.flytekit_async.wf?duration=all)
[wf2](https://development.uniondemo.run/console/projects/flytesnacks/domains/development/workflows/core.containerization.use_secrets.my_secret_workflow?duration=all)

[task 1](https://development.uniondemo.run/console/projects/flytesnacks/domains/development/tasks/async_prototype.flytekit_async.main?duration=all)
[task 2](https://development.uniondemo.run/console/projects/flytesnacks/domains/development/tasks/async_prototype.flytekit_async.get_data?duration=all)
[task 3](https://development.uniondemo.run/console/projects/flytesnacks/domains/development/tasks/async_prototype.flytekit_async.process_data?duration=all)

[launch plan](https://development.uniondemo.run/console/projects/flytesnacks/domains/development/launchPlans/async_prototype.flytekit_async.wf?duration=all)

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/3538 

## Follow-up issue
_NA_

### CODE REVIEW

File: packages/console/src/components/Entities/generators.ts [hunk 1]
```diff
@@ -16,56 +16,55 @@ export const executionFilterGenerator: {
...
   [ResourceType.DATASET]: noFilters,
-  [ResourceType.LAUNCH_PLAN]: ({ name }, version) =>
-    version
+  [ResourceType.LAUNCH_PLAN]: ({ name }, version) => [
+    {
+      key: 'launch_plan.name',
+      operation: FilterOperationName.EQ,
+      value: name,
+    },
+    ...(version
```
Comment (4nalog) [inline packages/console/src/components/Entities/generators.ts:25]: [nit] we can actually also use && to make it more concise

...(version && [...])

Comment (ursucarina) [reply packages/console/src/components/Entities/generators.ts:25]: i did it that way because there's a difference between
`[1,2, ...[]]` and `[1,2,...undefined]`
